test.js

```diff
@@ -525,133 +525,162 @@ test('text is right-aligned after wrapping', t => {
-	const longContent = sentence.repeat(3).trim();
-	const box = boxen(longContent, {align: 'right'});
+	const sentence = 'x'.repeat(width / 4) + ' ';
+	const longContent = sentence.repeat(4).trim();
+	const box = boxen(longContent);
 
-	t.is(box.length, width * 4);
+	const lines = box.split('\n');
 
-	const lines = [];
-	for (let index = 0; index < 4; ++index) {
-		const line = box.slice(index * width, (index + 1) * width);
-		t.is(line.length, width, 'Length of line #' + index);
-		t.is(line, line.trim(), 'No margin of line #' + index);
+	const checkAlign = (index, leftPad, rightPad) => {
+		const line = lines[index];
+		const lineWithoutBorders = line.slice(1, -1);
+		const paddingLeft = lineWithoutBorders.length - lineWithoutBorders.trimStart().length;
+		const paddingRight = lineWithoutBorders.length - lineWithoutBorders.trimEnd().length;
 
-		lines.push(line);
-	}
+		t.is(paddingLeft, leftPad, 'Padding left in line #' + index);
+		t.is(paddingRight, rightPad, 'Padding right in line #' + index);
+	};
 
-	const checkAlign = index => {
+	checkAlign(1, 0, 0);
+	checkAlign(2, 0, sentence.length * 2);
+});
+
+test('text is right-aligned after wrapping when using words', t => {
+	const width = process.stdout.columns || 120;
+	const sentence = 'x'.repeat(width / 4) + ' ';
+	const longContent = sentence.repeat(4).trim();
+	const box = boxen(longContent, {align: 'right'});
+
+	const lines = box.split('\n');
+
+	const checkAlign = (index, leftPad, rightPad) => {
 		const line = lines[index];
 		const lineWithoutBorders = line.slice(1, -1);
 		const paddingLeft = lineWithoutBorders.length - lineWithoutBorders.trimStart().length;
 		const paddingRight = lineWithoutBorders.length - lineWithoutBorders.trimEnd().length;
 
-		t.true(paddingLeft > 0, 'Padding left in line #' + index);
-		t.is(paddingRight, 0, 'Padding right in line #' + index);
+		t.is(paddingLeft, leftPad, 'Padding left in line #' + index);
+		t.is(paddingRight, rightPad, 'Padding right in line #' + index);
 	};
 
-	checkAlign(1);
-	checkAlign(2);
+	checkAlign(1, 0, 0);
+	checkAlign(2, sentence.length * 2, 0);
 });
 
 test('text is right-aligned after wrapping when using words, with padding', t => {
 	const width = process.stdout.columns || 120;
-	const sentence = 'x'.repeat(width / 3) + ' ';
-	const longContent = sentence.repeat(3).trim();
+	const sentence = 'x'.repeat(width / 4) + ' ';
+	const longContent = sentence.repeat(4).trim();
 	const box = boxen(longContent, {
 		align: 'right',
 		padding: {left: 1, right: 1, top: 0, bottom: 0}
 	});
 
-	t.is(box.length, width * 4);
-
-	const lines = [];
-	for (let index = 0; index < 4; ++index) {
-		const line = box.slice(index * width, (index + 1) * width);
-		t.is(line.length, width, 'Length of line #' + index);
-		t.is(line, line.trim(), 'No margin of line #' + index);
-
-		lines.push(line);
-	}
+	const lines = box.split('\n');
 
-	const checkAlign = index => {
+	const checkAlign = (index, leftPad, rightPad) => {
```

Comment:
```suggestion
	const checkAlign = ({index, leftPad, rightPad}) => {
```

To improve readability at the callsite.